enterprise/server/util/chunker/chunker.go

```diff
@@ -86,5 +81,10 @@ func New(ctx context.Context, averageSize int, writeChunkFn WriteFunc) (*Chunker
 		}
 	}()
 
+	go func() {
+		<-ctx.Done()
+		pw.Close()
```

Comment:
You should call CloseWithErr to be safe to avoid a scenario where the reader thinks the pipe was closed properly (i.e. something like `pw.CloseWithErr(context.Canceled)`